core/except.cpp

```diff
@@ -288,9 +288,31 @@ void validSlice(const Dimensions &dims, const Slice &slice) {
                              to_string(dims) + ".");
 }
 
-void coordsAndLabelsMatch(const DataConstProxy &a, const DataConstProxy &b) {
-  if (a.coords() != b.coords() || a.labels() != b.labels())
-    throw except::CoordMismatchError("Expected coords and labels to match.");
+void sparseCoordsAndLabelsMatch(const DataConstProxy &a,
+                                const DataConstProxy &b) {
+  if (b.dims().sparse()) {
+    const auto sparseDim = b.dims().sparseDim();
+
+    /* Fail if: */
+    /* - presence of a sparse dimension is not identical in both items */
```

Comment:
For the non-in-place operation it is supported to add a dense dataset to a sparse one (where the dense one lacks the sparse dimension), so this check is too restrictive?